### PR TITLE
Fix/plf 8512

### PR DESCRIPTION
**Pull request by mkrout**

### CODE REVIEW

File: portlets/src/main/frontend/src/apps/components/GamificationInformations/MybadgesInformations.vue [hunk 1]
```diff
@@ -36,7 +36,7 @@
                         </div>
 
                         <div class="box" v-else-if=" prog.value > badge.startScore || prog.value < badge.endScore ||prog.value < badge.startScore " :id="'rep'+badge.id">
-                            <img thumbnail fluid :id="'reputation'+badge.id" :src="`/rest/gamification/reputation/badge/${badge.title}/avatar`" alt="Thumbnail" class="greytwo" width="200px" height="200px" />
+                            <img thumbnail fluid :id="'reputation'+badge.id" :src="`/rest/gamification/reputation/badge/${badge.id}/avatar`" alt="Thumbnail" class="greytwo" width="200px" height="200px" />
```
Comment (ahamdi) [inline portlets/src/main/frontend/src/apps/components/GamificationInformations/MybadgesInformations.vue]:
```suggestion
                            <img thumbnail fluid :id="'reputation'+badge.id" :src="`/portal/rest/gamification/reputation/badge/${badge.id}/avatar`" alt="Thumbnail" class="greytwo" width="200px" height="200px" />
```

File: portlets/src/main/frontend/src/apps/components/badge/ManageBadges.vue [hunk 2]
```diff
@@ -121,8 +121,9 @@ export default {
                 .then(response => {
                     this.addSuccess = true;
                     this.updateMessage = 'updated';
-                    this.badges.push(badge);
+                   // this.badges.push(badgeDTO);
```
Comment (ahamdi) [inline portlets/src/main/frontend/src/apps/components/badge/ManageBadges.vue]: I think you can remove this line

File: portlets/src/main/frontend/src/apps/components/GamificationInformations/MybadgesInformations.vue [hunk 3]
```diff
@@ -125,7 +125,7 @@
                         </b-popover>
                     </div>
                     <div class="box" v-else-if=" prog.value > badge.startScore || prog.value < badge.endScore ||prog.value < badge.startScore " :id="'rep'+badge.id">
-                        <img thumbnail fluid :id="'reputation'+badge.id" :src="`/rest/gamification/reputation/badge/${badge.title}/avatar`" alt="Thumbnail" class="greytwo" width="200px" height="200px" />
+                        <img thumbnail fluid :id="'reputation'+badge.id" :src="`/rest/gamification/reputation/badge/${badge.id}/avatar`" alt="Thumbnail" class="greytwo" width="200px" height="200px" />
```
Comment (ahamdi) [inline portlets/src/main/frontend/src/apps/components/GamificationInformations/MybadgesInformations.vue]:
```suggestion
                        <img thumbnail fluid :id="'reputation'+badge.id" :src="`/portal/rest/gamification/reputation/badge/${badge.id}/avatar`" alt="Thumbnail" class="greytwo" width="200px" height="200px" />
```

File: portlets/src/main/frontend/src/apps/components/badge/BadgeList.vue [hunk 4]
```diff
@@ -185,7 +185,7 @@
                     <tbody>
                     <tr v-for="badge in filteredBadges">
                         <td id="iconInputGroup" style="max-width: 100px;">
-                            <div   style="z-index: 0;"> <img thumbnail fluid :src="`/rest/gamification/reputation/badge/${badge.title}/avatar`" alt="Thumbnail" class="m-1"  width="40" height="40"/>
+                            <div   style="z-index: 0;"> <img thumbnail fluid :src="`/rest/gamification/reputation/badge/${badge.id}/avatar`" alt="Thumbnail" class="m-1"  width="40" height="40"/>
```
Comment (ahamdi) [inline portlets/src/main/frontend/src/apps/components/badge/BadgeList.vue]:
```suggestion
                            <div   style="z-index: 0;"> <img thumbnail fluid :src="`/portal/rest/gamification/reputation/badge/${badge.id}/avatar`" alt="Thumbnail" class="m-1"  width="40" height="40"/>
```

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/BadgeService.java [hunk 5]
```diff
@@ -47,6 +47,28 @@ public BadgeDTO findBadgeByTitle(String badgeTitle) {
         }
         return null;
 
+    }
+    /**
+     * Find a BadgeEntity by id
+     * @param badgeTitle : badge id
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/BadgeService.java]: This is causing build failure for the PR
```suggestion
     * @param badgeId : badge id
```

File: portlets/src/main/frontend/src/apps/components/badge/ManageBadges.vue (hunk 2 above)
Comment (mkrout) [reply portlets/src/main/frontend/src/apps/components/badge/ManageBadges.vue]: yep

**Pull request merged**